Garage-sign documentation (#1728)
advancedtelematic/aktualizr: docs/ota-client-guide/modules/ROOT/pages/change-signature-thresholds.adoc

= Change signature thresholds
ifdef::env-github[]

[NOTE]
====
We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
====
endif::[]

> A threshold cryptosystem is one of the ways in which we protect your software repository. By default, the threshold number is one. This means that you can use any of the valid keys to sign your software versions. For more security, you can also change the threshold number of your Root and Targets keys, as well as delegations.

Reviewer: ```suggestion
One way you can keep your software repository more secure is by adding extra signing keys, and then setting up *key thresholds* to require metadata to be signed by multiple different keys. By default, each role on the repository has one signing key, and a threshold of one. For more security in the Root role, for example, you might want to create 5 different keys, keep them locked up at different locations around the world, and require that at least three of the keys sign off on any change to Root metadata. You can also use key thresholds for Targets metadata and for delegations.
```

Author: Changed

> *To change the thresholds of Root and Targets keys:*

Reviewer: ```suggestion
*To change the thresholds for signing Root and Targets metadata:*
```

> 

Reviewer: It doesn't really make sense to change the key thresholds without adding more keys. There should be a section here that explains how to add keys.

Author: Added.
Please review.

. In the *roles* folder of your local software repository, open the *unsigned* folder, and then open the `root.json` file.
. Depending on the threshold that you want to change, in the `signed` block, do the following:

** To change the threshold of Root keys:
.. In the `root` subsection, for the `keyids` object, specify the list of valid keys that can be used for signing.
+
You can find the list of all keys in the `keys` subsection.
+

> NOTE: The new version of Root metadata should be valid according to the rules of the previous and current versions. So you must sign Root metadata with a threshold of keys specified in the latest Root metadata file and a threshold of keys specified in the new Root metadata file. For more information, see the https://uptane.github.io/papers/uptane-standard.1.0.1.html#rfc.section.5.4.4.3[related section] of the Uptane standard.

Reviewer: ```suggestion
NOTE: The new version of Root metadata should be valid according to the rules of the previous and current versions. So you must sign Root metadata with the threshold of keys specified in the previous Root metadata file and the threshold of keys specified in the new Root metadata file. For more information, see the https://uptane.github.io/papers/uptane-standard.1.0.1.html#rfc.section.5.4.4.3[related section] of the Uptane standard.
```

Author: Changed.

.. For the `threshold` object, specify the threshold number.

** To change the threshold of Targets keys:
.. In the `targets` subsection, for the `keyids` object, specify the list of valid keys that can be used for signing.
+
You can find the list of all keys in the `keys` subsection.
.. For the `threshold` object, specify the threshold number.

. For the `version` object, specify the version number of the new Root metadata.

> . Sign the `root.json` file.

Reviewer: Since we're talking about adding extra keys and using thresholds, the example should include adding at least one more signature.
